rewrite-java-21/src/main/java/org/openrewrite/java/isolated/ReloadableJava21ParserVisitor.java

```diff
@@ -342,19 +342,25 @@ public J visitBreak(BreakTree node, Space fmt) {
     @Override
     public J visitCase(CaseTree node, Space fmt) {
         J.Case.Type type = node.getCaseKind() == CaseTree.CaseKind.RULE ? J.Case.Type.Rule : J.Case.Type.Statement;
+
+        JContainer<J> caselabels = JContainer.build(
+                node.getLabels().isEmpty() ? EMPTY : sourceBefore("case"),
+                node.getLabels().isEmpty() || node.getLabels().getFirst() instanceof DefaultCaseLabelTree ?
+                        List.of(JRightPadded.build(new J.Identifier(randomId(), Space.EMPTY, Markers.EMPTY, emptyList(), skip("default"), null, null))) :
+                        convertAll(node.getLabels(), commaDelim, t -> EMPTY),
```

Comment:
I suppose the override for `visitDeconstructionPattern()` will still be required. Is the plan to add that in this PR?

Reply:
Not yet, this PR aims to provide correct modelling of switch pattern matching and print idempotency for Record pattern matching (by mapping that to J.Unknown)